pkg/server/router.go

```diff
@@ -513,7 +513,7 @@ func verifyLocalPassword(ctx context.Context, config *Config, mu *sync.Mutex, de
 	mu.Lock()
 	defer mu.Unlock()
 
-	if _, ok := deferredNodes[node.Name]; !ok {
+	if _, ok := deferredNodes[node.Name]; !ok && !config.DisableAgent {
```

Comment:
I don't think here is a good place to do this either. There's no point in doing any of the stuff up above if this node isn't running an agent. Leave this line as-is, and just add a block up at the very top of this function:
```golang
	// do not attempt to verify the node password if the local host is not running an agent and does not have a node resource.
	if config.DisableAgent {
		return node.Name, http.StatusOK, nil
	}
```

Reply:
Done.
Thank you for your patience. I was too focused on my goal and didn't think about (slightly) larger picture 😅